lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -85,6 +85,14 @@ def add_node_data(node)
 
   # Rewrite a given file resource with the metadata from a fileserver based file
   def replace_metadata(resource, metadata)
+    if resource[:links] == "manage" && metadata.ftype == "link"
+      resource.delete(:source)
```

Comment:
Ah, my bad, my previous comment was on the commit, but @whopper explained why we *have* to delete `source` in this case: https://github.com/whopper/puppet/commit/6fc1f7da26c9bb7a9634fb7cec69ecd1f8013f0f#commitcomment-15871702. We'll talk about the "should we delete source in all cases" later today.